src/main/java/com/tictactoe/RulesFor3x3.java

```diff
@@ -31,17 +17,13 @@ public void identifyWinningCombos() {
 
 
     @Override
-    public boolean checkForWinner(Board board) {
+    public String checkForWinner(Board board) {
         for (int index = 0; index < this.winningCombos.length; index++) {
             if (threeInARow(board, index)) {
-                this.hasWinner = true;
-                this.winner = board.getSymbol(this.winningCombos[index][0]);
-                break;
-            } else {
-                this.hasWinner = false;
+                return board.getSymbol(this.winningCombos[index][0]);
```

Comment:
This line is quite messy. There is a lot of logic wrapped up in a small space. Maybe split it up into another method or a second line?